### PR TITLE
FEAT-#4622: Unify data type of log_level in logging module

### LINKED ISSUE #4622: Unify type of log_level in logging module
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. What kind of performance improvements would you like to see with this new API?

The logging module currently has a handful of functions that take in a `log_level` argument that are ints in some functions and string in other functions. For example, `_create_logger` in `logging/config.py` has `log_level` as an integer where as `class_logger.py` takes in `log_level` as a string. This isn't a terribly pressing issue, but we should try to decide on one type for `log_level` or handle both integers and strings elegantly.

### PR DESCRIPTION
**Pull request by arunjose696**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [ ] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #4622? 
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/logging/config.py [hunk 1]
```diff
@@ -36,6 +37,16 @@
 __LOGGER_CONFIGURED__: bool = False
 
 
+class LogLevel(IntEnum):  # noqa: PR01
+    """Enumerator to Specify the valid values of LogLevel accepted by Logger.setLevel()."""
```
Comment (YarShev) [inline modin/logging/config.py:41]:
```suggestion
    """Enumerator to specify the valid values of LogLevel accepted by Logger.setLevel()."""
```
Why noqa?

File: modin/test/test_logging.py [hunk 2]
```diff
@@ -27,8 +27,8 @@ def __init__(self, namespace):
         self.messages = collections.defaultdict(list)
         self.namespace = namespace
 
-    def info(self, message, *args, **kw):
```
Comment (YarShev) [inline modin/test/test_logging.py:30]: Why changed?

Comment (arunjose696) [reply modin/test/test_logging.py:30]: In previous impl, as log_level was a string the  logger was called as below, 

```
logger_level = getattr(logger, log_level)
logger_level(start_line)
```

now as log level is string this has been changed to call logger.log(log_level,message)

File: modin/logging/config.py (hunk 1 above)
Comment (arunjose696) [reply modin/logging/config.py:41]: was getting traceback  in doc_checker.py that {'value', 'names', 'start', 'type', 'module', 'qualname'} not documented, could see other intEnum using #noqa, so used this in this enum as well.

**Review by YarShev — APPROVED** [review]

**Pull request merged**